core/events.go

```diff
@@ -36,8 +42,40 @@ type ChainEvent struct {
 	Logs  []*types.Log
 }
 
+type ReOrgRollup struct {
+	ReOrgHeader     *types.Header
+	OldChainHeaders []*types.Header
+	NewChainHeaders []*types.Header
+}
 type ChainSideEvent struct {
 	Block *types.Block
 }
 
 type ChainHeadEvent struct{ Block *types.Block }
+
+// CopyHeader creates a deep copy of a block header to prevent side effects from
+// modifying a header variable.
+func CopyHeader(h *types.Header) *types.Header {
```

Comment:
Is there not another CopyHeader method we can access?

Reply:
I copied this method from somewhere else, but I will remove this code because now we are not using it at all